recipes/libdrm/all/conanfile.py

```diff
@@ -0,0 +1,172 @@
+        "shared": False,
+        "fPIC": True,
+        "libkms": True,
+        "intel": True,
+        "radeon": True,
+        "amdgpu": True,
+        "nouveau": True,
+        "vmwgfx": True,
+        "omap": False,
+        "exynos": False,
+        "freedreno": True,
+        "tegra": False,
+        "vc4": True,
+        "etnaviv": False,
+        "valgrind": False,
+        "freedreno-kgsl": False,
+        "udev": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("meson/0.59.0")
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.intel:
+            self.requires("libpciaccess/0.16")
+        if self.settings.os == "Linux":
+            self.requires("linux-headers-generic/5.14.9")
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            raise ConanInvalidConfiguration("libdrm supports only Linux or FreeBSD")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def _configure_meson(self):
+        meson = Meson(self)
+
+        defs={
+            "cairo-tests" : "false",
+            "install-test-programs": "false"
+        }
+        for o in ["libkms", "intel", "radeon", "amdgpu","nouveau", "vmwgfx", "omap", "exynos",
+                  "freedreno", "tegra", "vc4", "etnaviv", "valgrind", "freedreno-kgsl", "udev"]:
+            defs[o] = "true" if getattr(self.options, o) else "false"
+
+        defs["datadir"] = os.path.join(self.package_folder, "res")
+
+        meson.configure(
+            defs = defs,
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder)
+        return meson
+
+    def build(self):
+        meson = self._configure_meson()
+        meson.build()
+
+    def package(self):
+        meson = self._configure_meson()
+        meson.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.mkdir(os.path.join(self.package_folder, "licenses"))
+        # Extract the License/s from the header to a file
+        tmp = tools.load(os.path.join(self._source_subfolder, "include", "drm", "drm.h"))
+        license_contents = re.search("\*\/.*(\/\*(\*(?!\/)|[^*])*\*\/)", tmp, re.DOTALL)[1]
+        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), license_contents)
+
+    def package_info(self):
+        self.cpp_info.components["libdrm_libdrm"].libs = ["drm"]
```

Comment:
what's the reason for the `libdrm_` prefix in component names?

Reply:
this avoids collision with other recipes' components. Also, the component name cannot be identical to recipe name

Reply:
aren't components scoped? I mean to access a component you need to write `self.deps_cpp_info[package_name].components[component_name]`. 
can you point me to an example of such colission?
re. the second point, you can name the `"libdrm_libdrm` component `headers`.

Reply:
The components are scoped, but the files generated by generators (cmake_find_package etc.) are not